### PR TITLE
test: use Java 17 to produce Java 8-compatible bytecode

### PR DESCRIPTION
**Pull request by suztomo**
- To use Maven surefire plugin's "jvm" option to run the tests on Java 8. https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#jvm
- To use Java 17 to produce Java 8-compatible bytecode.

### CODE REVIEW

**Comment by suztomo** [comment]
"build(8) except for gapic-generator-java" was stuck:

```
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.168 s - in com.google.api.gax.grpc.GrpcLongRunningTest
[INFO] Running com.google.api.gax.grpc.GrpcDirectStreamControllerTest
Error: The operation was canceled.
```

Retrying.

File: gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ static ChannelPool create(ChannelPoolSettings settings, ChannelFactory channelFa
       throws IOException {
 
     // ProcessHandle was introduced in Java 9
-    System.out.println("ProcessHandle: " + java.lang.ProcessHandle.current());
+    System.out.println("org.graalvm.home.Version: " + org.graalvm.home.Version.parse("1.2.3"));
```
Comment (suztomo) [inline gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java:87]: When using `openjdk version "17.0.6" 2023-01-17`, the Maven test `sdk-platform-java$ mvn test -am -pl gax-java/gax-grpc` (without the jvm property) succeeds.

Comment (suztomo) [reply gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java:87]: The Java 8 check https://github.com/googleapis/sdk-platform-java/actions/runs/4982584325/jobs/8918462230?pr=1681 successfully detected this gap:

```
Error:  Errors: 
Error:    ChannelPoolTest.callShouldCompleteAfterCreation:254 » UnsupportedClassVersion org/graalvm/home/Version has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
Error:    ChannelPoolTest.callShouldCompleteAfterStarted:303 » UnsupportedClassVersion org/graalvm/home/Version has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

**Comment by suztomo** [comment]
"build(8) except for gapic-generator-java" passed: https://github.com/googleapis/sdk-platform-java/actions/runs/4982039825/jobs/8918034299

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -65,30 +65,19 @@ jobs:
...
-        shell: bash
-        run: |
-          set -x
-          export JAVA_HOME=$JAVA8_HOME
-          export PATH=${JAVA_HOME}/bin:$PATH
-          # useIncrementalCompilation=false to avoid recompiling classes
-          # generated by Java 11 compiler above.
-          mvn -B -ntp test  --projects '!gapic-generator-java' \
-              -Dcheckstyle.skip -Dmaven.compiler.useIncrementalCompilation=false \
-              -Dfmt.skip
```
Comment (suztomo) [inline .github/workflows/ci.yaml:91]: Note: We used to run two Maven commands, one to compile and the other to run tests. However, the same thing can be done in surefire's jvm system property.

File: .kokoro/release/common.cfg [hunk 3]
```diff
@@ -9,7 +9,7 @@ build_file: "gapic-generator-java/.kokoro/trampoline.sh"
 # Configure the docker image for kokoro-trampoline.
 env_vars: {
   key: "TRAMPOLINE_IMAGE"
-  value: "gcr.io/cloud-devrel-kokoro-resources/java11"
+  value: "gcr.io/cloud-devrel-kokoro-resources/java17"
```
Comment (suztomo) [inline .kokoro/release/common.cfg:12]: We'll produce Java 8-compatible bytecode using Java 17.

**Review by meltsufin — APPROVED** [review]

**Comment by suztomo** [comment]
Confirming without a real release. Build ID b0a899d7-ed51-4b8b-b32b-0fb91876eac6

**Comment by suztomo** [comment]
I confirmed the generated class files are for Java 8.

```
suztomo@suztomo2:~/sdk-platform-java$ javap -v -cp ~/.m2/repository/com/google/api/gapic-generator-java/2.19.1-SNAPSHOT/gapic-generator-java-2.19.1-SNAPSHOT.jar com/google/api/generator/Main |head
Classfile jar:file:///usr/local/google/home/suztomo/.m2/repository/com/google/api/gapic-generator-java/2.19.1-SNAPSHOT/gapic-generator-java-2.19.1-SNAPSHOT.jar!/com/google/api/generator/Main.class
  Last modified May 22, 2023; size 1754 bytes
  SHA-256 checksum bdb9b512c7a2ddc82beb46de8b47f47b9d825028d54d49c851f077f206ec529a
  Compiled from "Main.java"
public class com.google.api.generator.Main
  minor version: 0
  major version: 52
  flags: (0x0021) ACC_PUBLIC, ACC_SUPER
  this_class: #47                         // com/google/api/generator/Main
  super_class: #2                         // java/lang/Object
```

**Pull request merged**